src/Materializer/RecordModelFactory.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Kros.KORM.Converter;
+using Kros.KORM.Injection;
+using Kros.KORM.Metadata;
+using System;
+using System.Data;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace Kros.KORM.Materializer
+{
+    /// <summary>
+    /// Model factory for record types.
+    /// </summary>
+    internal static class RecordModelFactory
+    {
+        public static Func<IDataReader, T> CreateFactoryForRecords<T>(
+           IDataReader reader,
+           TableInfo tableInfo,
+           IInjector injector,
+           ConstructorInfo ctor)
+        {
+            Type type = typeof(T);
+            var dynamicMethod = new DynamicMethod(
+                $"korm_factory_record_{typeof(T).Name}",
+                type, new Type[] { typeof(IDataReader) }, true);
+            ILGenerator iLGenerator = dynamicMethod.GetILGenerator();
+            ParameterInfo[] paramsInfo = ctor.GetParameters();
+
+            foreach (ParameterInfo param in paramsInfo)
+            {
+                if (injector.IsInjectable(param.Name))
+                {
+                    iLGenerator.CallGetInjectedValue(injector, param.Name, param.ParameterType);
+                }
+                else
+                {
+                    ColumnInfo columnInfo = tableInfo.GetColumnInfoByPropertyName(param.Name);
```

Comment:
Can it happen, that constructor has a parameter, which does not match a property name? In that case `columnInfo` would be`null`.